app/src/main/java/com/example/appmarvel/activity/MainActivity.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.appmarvel.activity
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.example.appmarvel.databinding.ActivityMainBinding
+import com.example.appmarvel.mvvm.viewmodel.MainViewModel
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class MainActivity : AppCompatActivity(), KoinComponent {
+
+    private val viewModel: MainViewModel by inject()
+    private lateinit var binding: ActivityMainBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        setListeners()
+        viewModel.listenerState.observe({ lifecycle }, ::changeStateListener)
+    }
+
+    private fun changeStateListener(listenerData: MainViewModel.ListenerData) {
+        if (listenerData.lState == MainViewModel.ListenerState.GO_TO_APP_PRESSED) {
```

Comment:
this should be a when statement, so when you add more states in the future you will not need to update the whole method

Reply:
Done